File: camerax_android/android/src/main/kotlin/dev/hebei/camerax_android/core/SurfaceOrientedMeteringPointFactoryImpl.kt
package dev.hebei.camerax_android.core

import dev.hebei.camerax_android.CameraXImpl
import dev.hebei.camerax_android.PigeonApiSurfaceOrientedMeteringPointFactoryApi

class SurfaceOrientedMeteringPointFactoryImpl(impl: CameraXImpl) :
    PigeonApiSurfaceOrientedMeteringPointFactoryApi(impl) {
    override fun pigeon_defaultConstructor(
        width: Double, height: Double
    ): androidx.camera.core.SurfaceOrientedMeteringPointFactory {
        return androidx.camera.core.SurfaceOrientedMeteringPointFactory(width.toFloat(), height.toFloat())
    }
}

File: camerax_android/lib/src/core/surface_oriented_metering_point_factory_impl.dart
import 'package:camerax_android/src/api.dart';
import 'package:camerax_android/src/core.dart';
import 'package:camerax_platform_interface/camerax_platform_interface.dart';

final class SurfaceOrientedMeteringPointFactoryImpl
    extends MeteringPointFactoryImpl
    implements SurfaceOrientedMeteringPointFactory {
  @override
  final SurfaceOrientedMeteringPointFactoryProxyApi api;

  SurfaceOrientedMeteringPointFactoryImpl.internal(this.api);
}

final class SurfaceOrientedMeteringPointFactoryChannelImpl
    extends SurfaceOrientedMeteringPointFactoryChannel {
  @override
  SurfaceOrientedMeteringPointFactory create(double width, double height) {
    final api = SurfaceOrientedMeteringPointFactoryProxyApi(
      width: width,
      height: height,
    );
    return SurfaceOrientedMeteringPointFactoryImpl.internal(api);
  }
}
